hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/operations/CustomCallCodeOperationTest.java

```diff
@@ -0,0 +1,109 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.test.exec.operations;
+
+import static com.hedera.node.app.service.contract.impl.exec.failure.CustomExceptionalHaltReason.MISSING_ADDRESS;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.REQUIRED_GAS;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.assertSameResult;
+import static org.hyperledger.besu.evm.frame.ExceptionalHaltReason.INSUFFICIENT_GAS;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doCallRealMethod;
+
+import com.hedera.node.app.service.contract.impl.exec.AddressChecks;
+import com.hedera.node.app.service.contract.impl.exec.operations.CustomCallCodeOperation;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.internal.FixedStack;
+import org.hyperledger.besu.evm.operation.Operation;
+import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CustomCallCodeOperationTest {
+
+    @Mock
+    private GasCalculator gasCalculator;
+
+    @Mock
+    private AddressChecks addressChecks;
+
+    @Mock
+    private MessageFrame frame;
+
+    @Mock
+    private WorldUpdater worldUpdater;
+
+    @Mock
+    private EVM evm;
+
+    private CustomCallCodeOperation subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new CustomCallCodeOperation(gasCalculator, addressChecks);
+    }
+
+    @Test
+    void catchesUnderflowWhenStackIsEmpty() {
+        given(frame.getStackItem(1)).willThrow(FixedStack.UnderflowException.class);
+        final var expected = new Operation.OperationResult(0L, ExceptionalHaltReason.INSUFFICIENT_STACK_ITEMS);
+        assertSameResult(expected, subject.execute(frame, evm));
+    }
+
+    @Test
+    void rejectsMissingNonSystemAddress() {
+        doCallRealMethod().when(addressChecks).isNeitherSystemNorPresent(any(), any());
+        givenWellKnownFrameWith(1L, Address.fromHexString("0x123"), 2L);
+        final var expected = new Operation.OperationResult(123L, MISSING_ADDRESS);
```

Comment:
wait, why is it `123` base **16** on line 81 and `123` base **10** on line 82?  Is that just to emphasize that, "yes, definitely _missing_!"?  Or just to catch out unwary readers?  (I was paying attention!)

Reply:
One is an address and one a gas cost; replaced with `NON_SYSTEM_LONG_ZERO_ADDRESS` and `REQUIRED_GAS`, respectively, to make this readable.